FirebaseDatabase/Sources/Utilities/FUtilities.m

```diff
@@ -295,10 +295,11 @@ + (NSError *)errorForStatus:(NSString *)status andReason:(NSString *)reason {
 }
 
 + (NSNumber *)intForString:(NSString *)string {
-    static NSCharacterSet *notDigits = nil;
-    if (!notDigits) {
-        notDigits = [[NSCharacterSet decimalDigitCharacterSet] invertedSet];
-    }
+    static dispatch_once_t once;
+    static NSCharacterSet *notDigits;
+    dispatch_once(&once, ^{
+      notDigits = [[NSCharacterSet decimalDigitCharacterSet] invertedSet];
+    });
```

Comment:
Curious, what is the reason we want to assign the `notDigits` just once? Is it expensive to `[[NSCharacterSet decimalDigitCharacterSet] invertedSet]`?

Reply:
I was focused on correcting the existing implementation - but presumably it is converting a operation processing the Unicode character set to a cached value which could add up if this function is called a lot. cc: @schmidt-sebastian

Reply:
Gotcha, interesting, makes sense. Nice fix!